opensquirrel/common.py

```diff
@@ -28,7 +28,7 @@ def normalize_angle(x: SupportsFloat) -> float:
 
 
 def are_matrices_equivalent_up_to_global_phase(
-    matrix_a: NDArray[np.complex128], matrix_b: NDArray[np.complex128]
+    matrix_a: NDArray[np.complex128], matrix_b: NDArray[np.complex128 | np.float64]
```

Comment:
I understand we are adding this `np.float64` because `np.eye(matrix.shape[0])` uses `np.float64` by default. But we can also keep this function as it is and use `np.eye(matrix.shape[0], dtype=np.complex128)`. Which makes more sense, because quantum matrices are supposed to have complex elements.

```suggestion
    matrix_a: NDArray[np.complex128], matrix_b: NDArray[np.complex128]
```